adapter/repository/in-memory-simple/src/test/java/org/mobilitydata/gtfsvalidator/db/NoticeMemoryConsumptionTest.java

```diff
@@ -0,0 +1,140 @@
+package org.mobilitydata.gtfsvalidator.db;
+
+import org.junit.jupiter.api.Test;
+import org.mobilitydata.gtfsvalidator.usecase.notice.IntegerFieldValueOutOfRangeNotice;
+import org.mobilitydata.gtfsvalidator.usecase.notice.base.ErrorNotice;
+import org.mobilitydata.gtfsvalidator.usecase.notice.base.InfoNotice;
+import org.mobilitydata.gtfsvalidator.usecase.notice.base.Notice;
+import org.mobilitydata.gtfsvalidator.usecase.notice.base.WarningNotice;
+import org.mobilitydata.gtfsvalidator.usecase.port.ValidationResultRepository;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class NoticeMemoryConsumptionTest {
+
+    private void generateNotices(MockValidationResultRepo mockValidationResultRepo, int numberOfNotices) {
```

Comment:
If you're trying to ensure that the validator itself doesn't throw an exception in production on a certain number of notices, then this method should use the actual repo implementation being used in production, not a mocked repo.

Reply:
Nice catch! I'm adapting the code to your advice. Thanks